gateway-js/src/index.ts

```diff
@@ -230,17 +231,22 @@ export class ApolloGateway implements GraphQLService {
 
     this.experimental_pollInterval = config?.experimental_pollInterval;
 
-    // Do not use this unless advised by Apollo staff to do so
-    // 1. If config is explicitly set to a `string` or `null` in code, use it
-    // 2. Else, if the env var is set, use that
-    // 3. Else, default to `null`
-    this.experimental_schemaConfigDeliveryEndpoint = isPrecomposedManagedConfig(
-      this.config,
-    )
-      ? this.config.experimental_schemaConfigDeliveryEndpoint
-      : this.config.experimental_schemaConfigDeliveryEndpoint === null
-      ? null
-      : process.env.APOLLO_SCHEMA_CONFIG_DELIVERY_ENDPOINT ?? null;
+    // 1. If config is set to a `string`, use it
+    // 2. If config is `undefined`, use the default uplink URL
+    // 3. If config is explicitly set to `null`, fallback to GCS
+    // 4. If the env var is set, use that
+    this.schemaConfigDeliveryEndpoint = 'https://uplink.api.apollographql.com/';
+
+    // This if case unobviously handles 1, 2, and 4.
+    if (isPrecomposedManagedConfig(this.config)) {
+      const envEndpoint = process.env.APOLLO_SCHEMA_CONFIG_DELIVERY_ENDPOINT;
+      this.schemaConfigDeliveryEndpoint =
+        this.config.schemaConfigDeliveryEndpoint ??
+        envEndpoint ??
+        this.schemaConfigDeliveryEndpoint;
```

Comment:
Why write it this way instead of just setting the default here?  ie why do we want `this.schemaConfigDeliveryEndpoint` to be set at all when not using managed?

Reply:
Ah yeah - just needed to make it optional on the class as well which resulted in no type errors, cool
9f9a2b06b8bec3e5f2cbe2038c96036c5b0ed22b